internal/rbd/controllerserver.go

```diff
@@ -368,6 +371,20 @@ func flattenTemporaryClonedImages(ctx context.Context, rbdVol *rbdVolume, cr *ut
 		}
 		return status.Errorf(codes.ResourceExhausted, "rbd image %s has %d snapshots", rbdVol, len(snaps))
 	}
+
+	if len(snaps) > int(minSnapshotsOnImageToStartFlatten) {
```

Comment:
The soft limit should flatten to only reduce the outstanding count by a value that lowers it below the soft limit, and not all snapshots till the soft limit.

In this case, about 250 snapshots would be flattened, which would take time. The time taken would be lower than the 450 max limit, but still the operation would not be fast enough.

Instead, if we flatten only as many snapshots as required to go below the soft limit, then the repetitive operation cost is amortized across future clones, and each clone operation hence would only be slowed down as much.

So I would suggest we either pass the delta count to `flattenClonedRbdImages` and only flatten as many, or pass in `snaps` list with only a count of snaps that require flattening.

Reply:
updated logic to flatten only a minimum number of snapshots once the limit is reached so that the volume creation time should not be affected heavily